core/src/main/java/io/seata/core/protocol/Version.java

```diff
@@ -69,8 +76,35 @@ public static String getChannelVersion(Channel c) {
      * @return the string
      * @throws IncompatibleVersionException the incompatible version exception
      */
-    public static String checkVersion(String version) throws IncompatibleVersionException {
-        // TODO: check
-        return version;
+    public static void checkVersion(String version) throws IncompatibleVersionException {
+        long current = convertVersion(CURRENT);
+        long clientVersion = convertVersion(version);
+        long divideVersion = convertVersion(VERSION_0_7_1);
+        if ((current - divideVersion > 0 && clientVersion - divideVersion < 0) || (current - divideVersion < 0 && clientVersion - divideVersion > 0)) {
```

Comment:
Is it clearer to change 【current - divideVersion > 0】 to 【current > divideVersion】

Reply:
fixed